core/repository/block/index.ts

```diff
@@ -13,112 +10,54 @@ export interface IBlockRepository extends IBlockRepositoryShared {
 
 class BlockRepo implements IBlockRepository {
     private memoryBlocks: Array<Block> = [];
-    private lastBlock: Block = null;
-    private lastNBlockIds: Array<BlockId> = [];
 
     public add(block: Block): Block {
         this.memoryBlocks.push(block);
-        return block;
-    }
-
-    public delete(block: Block): DeletedBlockId {
-        for (let i = 0; i < this.memoryBlocks.length; i++) {
-            if (this.memoryBlocks[i].id === block.id) {
-                this.memoryBlocks.splice(i, 1);
-                break;
-            }
-        }
-        return block.id;
-    }
-
-    public deleteAfterBlock(blockId: string): void {
-        let index = null;
-        for (let i = 0; i < this.memoryBlocks.length; i++) {
-            if (this.memoryBlocks[i].id === blockId ) {
-                index = i;
-                break;
-            }
-        }
-        this.memoryBlocks = this.memoryBlocks.slice(0, index + 1);
-        return;
-    }
 
+        messageON('LAST_BLOCKS_UPDATE', {
+            lastBlock: block
+        });
 
-    getById(blockId: BlockId): Block {
-        for (let i = 0; i < this.memoryBlocks.length; i++) {
-            if (this.memoryBlocks[i].id === blockId) {
-                return this.memoryBlocks[i].getCopy();
-            }
-        }
-        return;
+        return block;
     }
 
     public getGenesisBlock(): Block {
-        return this.memoryBlocks[0].getCopy();
+        return this.memoryBlocks[0];
     }
 
     public getLastBlock(): Block {
-        return this.lastBlock;
+        return this.memoryBlocks.length ? this.memoryBlocks[this.memoryBlocks.length - 1] : null;
     }
 
     public deleteLastBlock(): Block {
-        this.memoryBlocks.length = this.memoryBlocks.length - 1;
-        this.lastBlock = this.memoryBlocks[this.memoryBlocks.length - 1];
+        this.memoryBlocks.pop();
```

Comment:
Should we be able to remove a genesis block?

Reply:
Yes, function in repository independent of logic (like genesisBlock)